### PR TITLE
Added 400 response

### PR DESCRIPTION
**Pull request by os6sense**
- added missing specs
- changed spec description and code for unknown errors

### CODE REVIEW

File: lib/hammer/request.rb [hunk 1]
```diff
@@ -54,6 +54,8 @@ def response
 
     def raise_error
       case response.code
+      when 400
+        raise Errors::Request400Error.new(url, method, params)
```
Comment (cheba) [inline lib/hammer/request.rb]: `Request400Error` is such a bad name…

**Comment by cheba** [comment]
:star:

**Comment by alaibe** [comment]
This PR already exists: https://github.com/smartpension/hammer/pull/16/files

File: lib/hammer/request.rb (hunk 1 above)
Comment (alaibe) [inline lib/hammer/request.rb]: What do you suggest?

Comment (cheba) [inline lib/hammer/request.rb]: `Errors::BadRequest` for instance.

**Comment by alaibe** [comment]
Could you describe the need?

**Comment by os6sense** [comment]
@alaibe: Yes, I'd added it in and then saw Alessandro had a PR pending with the exact same changes.

The need is that if request is submitted to the API with malformed syntax it currently simply returns the generic form or RequestError; 400 is quite a specific one though and it would be very useful to be able to differentiate in spanner since it is a common enough response.

**Comment by alaibe** [comment]
The need is that if request is submitted to the API with malformed syntax it currently simply returns the generic form or RequestError; 400 is quite a specific one though and it would be very useful to be able to differentiate in spanner since it is a common enough response.
So you catch it somewhere?

**Comment by os6sense** [comment]
I’m not catching it at the moment since the submission is throwing a generic error. I’d like to have the option to catch it and know that I was dealing with malformed request syntax over something more generic. 400 provides quite a lot of information ..

**Comment by alaibe** [comment]
:-1: until I see code who use this.

**Pull request closed without merging**